neqo-http3/src/connection_client.rs

```diff
@@ -5562,4 +5656,311 @@ mod tests {
+        // Connect and send a request
+        let (mut client, mut server, request_stream_id) = connect_and_send_request(true);
+
+        // Send a push promise.
+        send_push_promise(&mut server.conn, request_stream_id, FIRST_PUSH_ID);
+
+        let headers1xx = vec![(String::from(":status"), String::from("101"))];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers1xx, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        let mut d = Encoder::default();
+        hframe.encode(&mut d);
+        let headers200 = vec![
+            (String::from(":status"), String::from("200")),
+            (String::from("my-header"), String::from("my-header")),
+            (String::from("content-length"), String::from("3")),
+        ];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers200, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        hframe.encode(&mut d);
+
+        // create a push stream.
+        let _ = send_push_with_data(
+            &mut server.conn,
+            u8::try_from(FIRST_PUSH_ID).unwrap(),
+            &d,
+            true,
+        );
+
+        server_send_response_and_exchange_packet(
+            &mut client,
+            &mut server,
+            request_stream_id,
+            HTTP_RESPONSE_2,
+            true,
+        );
+
+        let mut events = client.events();
+
+        let (push_id_1xx, interim_1xx, headers_1xx) = events
+            .find_map(|e| {
+                if let Http3ClientEvent::PushHeaderReady {
+                    push_id,
+                    interim,
+                    headers,
+                    ..
+                } = e
+                {
+                    Some((push_id, interim, headers))
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+        assert_eq!(push_id_1xx, FIRST_PUSH_ID);
+        assert!(interim_1xx);
+        assert_eq!(headers_1xx, headers1xx);
+
+        let (push_id_200, interim_200, headers_200) = events
+            .find_map(|e| {
+                if let Http3ClientEvent::PushHeaderReady {
+                    push_id,
+                    interim,
+                    headers,
+                    ..
+                } = e
+                {
+                    Some((push_id, interim, headers))
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+        assert_eq!(push_id_200, FIRST_PUSH_ID);
+        assert!(!interim_200);
+        assert_eq!(headers_200, headers200);
+    }
+
+    // Client: receive a push stream
+    #[test]
+    fn push_single_wo_status() {
```

Comment:
Is there another test here?  A PushPromise that doesn't have `:authority` or similar should do the same, right?

Reply:
This is still missing #753